pkg/reconciler/delivery/delivery.go

```diff
@@ -99,53 +116,128 @@ func (c *Reconciler) updateRoute(ctx context.Context, cfg *v1.Configuration) err
 	if err != nil {
 		return err
 	}
+
 	logger.Info("Applying updated Route object")
+
+	// record the timestamp for the current udpate to the Route object before actually pushing it
+	// this is used later when determining if Route status is up to date
+	if route.Annotations == nil {
+		route.Annotations = make(map[string]string)
+	}
+	route.Annotations[AnnotationKey] = c.timeProvider().Format(TimeFormat)
 	_, err = c.client.ServingV1().Routes(cfg.Namespace).Update(route)
+	if err != nil {
+		return err
+	}
 
-	return err
+	// when we have latestRevision = true, we know that we don't need to queue future events
+	if *route.Spec.Traffic[0].LatestRevision {
+		logger.Info("Progressive rollout completed!")
+		return nil
+	}
+	t, e := getThreshold(&policy, int(*route.Spec.Traffic[1].Percent))
+	if e != nil {
+		return e
+	}
+	logger.Infof("Queueing event for %v seconds later", t)
+	c.followup(cfg, time.Duration(t) * time.Second)
+
+	return nil
 }
 
 // isRouteStatusUpToDate determines if the current Route status already matches our desired state
 func isRouteStatusUpToDate(route *v1.Route, newRevName string) bool {
```

Comment:
not a blocker but a lot of these functions should have unit tests.

Reply:
I will write unit tests for these in the next PR